src/mono/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -100,6 +96,23 @@ static string CombineSearchPath(NSSearchPathDirectory searchPath, string subdire
                     Path.Combine(path, subdirectory) :
                     string.Empty;
             }
+
+            static string CombineDocumentDirectory(string subdirectory)
+            {
+#if TARGET_TVOS
+                string? path = CombineSearchPath(NSSearchPathDirectory.NSLibraryDirectory, Path.Combine("Caches", "Documents", subdirectory));
+                // Special version of CombineSearchPath which creates the path if needed.
+                // This isn't needed for "real" search paths which always exist, but on tvOS
+                // the base path is really a subdirectory we define rather than an OS directory.
+                // In order to not treat Directory.Exists(SpecialFolder.ApplicationData) differently
+                // on tvOS, guarantee that it exists by creating it here
+                if (!Directory.Exists (path))
+                    Directory.CreateDirectory (path);
```

Comment:
Nit, but if this can't be created for some reason then Environment.GetFolderPath will throw an exception it isn't currently documented to throw.

Reply:
What do you feel would be the correct approach? Don't attempt to create the folder?  Document the possible exception?